indexer.ts

```diff
@@ -0,0 +1,743 @@
+      tx_hash: receiptId,
+    };
+
+    await context.db.Activity.insert(activity);
+  }
+
+  async function handleProjectRegistrationUpdate(
+    args,
+    receiverId,
+    signerId,
+    receiptId
+  ) {
+    let data = JSON.parse(args);
+    console.log("new Project data::", { ...data }, receiverId);
+    let regUpdate = {
+      status: data.status,
+      admin_notes: data.review_notes,
+      updated_at: new Date(
+        Number(BigInt(block.header().timestampNanosec) / BigInt(1000000))
+      ),
+    };
+
+    await context.db.ListRegistration.update(
+      { registrant_id: data.project_id },
+      regUpdate
+    );
+  }
+
+  async function handlePotApplication(args, receiverId, signerId, receiptId) {
+    let data = JSON.parse(args);
+    console.log("new factory data::", { ...data }, receiverId);
+    await context.db.Account.upsert({ id: data.project_id }, ["id"], []);
+    let application = {
+      pot_id: receiverId,
+      applicant_id: data.project_id,
+      message: data.message,
+      submitted_at: data.submitted_at,
+      current_status: data.status,
+      tx_hash: receiptId,
+    };
+    const appl = await context.db.PotApplication.insert(application);
+
+    let activity = {
+      signer_id: signerId,
+      receiver_id: receiverId,
+      timestamp: application.submitted_at,
+      type: "Submit_Application",
+      action_result: appl[0].id.toString(), // result points to the pot application created.
+      tx_hash: receiptId, // should we have receipt on both action and activity?
+    };
+
+    await context.db.Activity.insert(activity);
+  }
+
+  async function handleApplicationStatusChange(
+    args,
+    receiverId,
+    signerId,
+    receiptId
+  ) {
+    let data = JSON.parse(args);
+    console.log("new factory data::", { ...data }, receiverId);
+
+    let receipt = block
+      .receipts()
+      .filter((receipt) => receipt.receiptId == receiptId)[0];
+    let update_data = JSON.parse(atob(receipt.status["SuccessValue"]));
+    let appl = (
+      await context.db.PotApplication.select({ applicant_id: data.project_id })
+    )[0];
+
+    let applicationReview = {
+      application_id: appl.id,
+      reviewer_id: signerId,
+      notes: update_data.notes,
+      status: update_data.status,
+      reviewed_at: update_data.updated_at,
+    };
+    let applicationUpdate = {
+      current_status: update_data.status,
+      last_updated_at: update_data.updated_at,
+    };
+
+    await context.db.PotApplicationReview.insert(applicationReview);
+    await context.db.PotApplication.update({ id: appl.id }, applicationUpdate);
+  }
+
+  async function handleDefaultListStatusChange(
+    args,
+    receiverId,
+    signerId,
+    receiptId
+  ) {
+    let data = JSON.parse(args);
+    console.log("update project data::", { ...data }, receiverId);
+
+    let listUpdate = {
+      default_registration_status: data.status,
+    };
+
+    await context.db.List.update({ id: receiverId }, listUpdate);
```

Comment:
this works for the current registry contract but not for Lists contract, where the list_id will be specified in the call. changing to `id: data.list_id || receiverId` should resolve